src/ripple/protocol/tokens.h

```diff
@@ -90,24 +96,83 @@ base58EncodeTokenBitcoin (TokenType type, void const* token, std::size_t size);
+
+    Ripple lib encoded seeds start with a three-byte prefix of:
+    <TokenType::None><0xE1><0x4B> rather than the usual one-byte prefix of:
+    <TokenType::FamilySeed>
+*/
+enum class ExtraB58Encoding {None, RippleLib};
+
+/** Decode a base58 family seed.
+
+    Return `boost::none` if the encoding could not be interpreted as a family seed.
+    return the extra encoding type if the encoding is a family seed. The extra encoding
+    type is either `RippleLib` for ripple lib encoded seeds (these are ed25519 seeds with
+    a special prefix) to `None` for regular seeds.
+*/
+boost::optional<ExtraB58Encoding>
+decodeBase58FamilySeed(Slice s, MutableSlice result);
 
 /** Decode a Base58 token using Bitcoin alphabet
 
-    The type and checksum must match or an
-    empty string is returned.
+   The type and checksum must match or `boost::none is returned``. The slice
+   must decode into exactly as many bytes as specified as the result slice or
+   `false` is returned. The value will be decoded into the slice specified
+   by the `result` parameter.
 
-    This is used to detect user error. Specifically,
-    when an AccountID is specified using the wrong
-    base58 alphabet, so that a better error message
-    may be returned.
+   @note This is used to detect user error. Specifically, when an AccountID is
+   specified using the wrong base58 alphabet, so that a better error message may
+   be returned.
 */
-std::string
-decodeBase58TokenBitcoin(std::string const& s, TokenType type);
+bool
+decodeBase58TokenBitcoin(
+    Slice s,
+    TokenType type,
+    MutableSlice result);
+
+/** Metadata associated with an encoding.
+
+    Tokens are encoded as:
+    <1-byte TokenType><Optional 2-byte Encoding type for ripple-lib><Data><4-byte checksum>
+    The metadata includes the non-data part of the encoding. If an encoding does not include
+    an encoding type, the metadata will use 2-bytes of zeros.
+*/
+struct DecodeMetadata
+{
+    std::array<std::uint8_t, 2> encodingType;
+    std::uint8_t tokenType;
+    std::array<std::uint8_t, 4> checksum;
+    bool isRippleLibEncoded() const;
+};
+
+/** Low-level decode routine. This can be used to when the token type is unknown. If the
+    Token type is known, use either `decodeBase58Token` or `decodeBase58FamilySeed`.
+
+    The checksum must match or `boost::none` is returned. The value will be
+    decoded into the slice specified by the `result` parameter. If `allowResize`
+    is true, the result may be smaller than the specified slice. In that case
+    the returned slice will be a proper subset of the result slice. If
+    `allowResize` is false, the returned slice is always either `boost::none` or
+    is the same slice as `result`. If the result is larger than the space
+    allowed by `result` then `boost::none` is returned.
+*/
+boost::optional<std::pair<Slice, DecodeMetadata>>
+decodeBase58(
+    Slice s,
+    MutableSlice result,
+    bool allowResize);
```

Comment:
would prefer an enum over a bool parameter. Otherwise you have to add comments like `..., /*allowResize*/ true)` at the callsite. 

vs. something like `enum class decodeBase58Options { None, AllowResize };` then `..., decodeBase58Options::AllowResize);`

No more comment needed, self documenting, easy to add more options (or turn it into a bit field).

Reply:
Addressed in 91c6c712b0973af30980b849d50e33a4f715d449
Note: This was addressed by removing the `allowResize` param altogether. This low-level routine was renamed to `decodeBase58Resizable` and it always allows resizing. It was only called once (apart from testing); Most code uses the higher-leve token decoding functions which never allow resizing.